src/test/java/net/sf/jabref/logic/cleanup/CleanupWorkerTest.java

```diff
@@ -54,10 +55,17 @@ public void setUp() throws IOException {
         MetaData metaData = new MetaData();
         metaData.setDefaultFileDirectory(pdfFolder.getAbsolutePath());
         BibDatabaseContext context = new BibDatabaseContext(new BibDatabase(), metaData, bibFolder.newFile("test.bib"));
+
+        JabRefPreferences prefs = JabRefPreferences.getInstance();
+        prefs.putBoolean(JabRefPreferences.BIB_LOC_AS_PRIMARY_DIR, true);
```

Comment:
This really changes the preferences of the developer. Can you try to mock/init by hand the layout formatter and file directory preferences.